Update Runs API for new saved plan runs functionality (#428)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/run.mdx

| Operation        | Description                                                                                                                                                                |
|------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `plan_only`      | The run does not have an apply phase. This is also called a [speculative plan](/terraform/cloud-docs/run/modes-and-options#plan-only-speculative-plan).                |
| `plan_and_apply` | The run includes both plan and apply phases.                                                                                                               |

> | `save_plan`      | The run is a saved plan run; it can include both plan and apply phases, but won't become the workspace's current run unless it is confirmed for apply.                     |

Reviewer: Small suggestion for active voice and to avoid using a semicolon outside of code:
```suggestion
| `save_plan`      | The run is a saved plan run and can include both plan and apply phases, but it will only become the workspace's current run if you confirm you want to apply it.                     |
```

Author: Aww, but I like my semis 😂 —hmm, I'm not a fan of "and...but" here, so let's just split it to sentences.

| `data.attributes.target-addrs`                     | array\[string]       |                                                    | Specifies an optional list of resource addresses to be passed to the `-target` flag.                                                                                                                                                          |
| `data.attributes.variables`                        | array\[{key, value}] | (empty array)                                      | Specifies an optional list of run-specific variable values. Refer to [Run-Specific Variables](/terraform/cloud-docs/workspaces/variables/managing-variables#run-specific-variables) for details.    |
| `data.attributes.plan-only`                        | bool                 | (from configuration version)                       | Specifies if this is a [speculative, plan-only](/terraform/cloud-docs/run/modes-and-options#plan-only-speculative-plan) run that Terraform cannot apply. Often used in conjunction with terraform-version in order to test whether an upgrade would succeed. |

> | `data.attributes.save-plan`                        | bool                 | false                                              | Specifies if this should be a saved plan run. Saved plan runs perform their plan and checks without becoming the workspace's current run (like a plan-only run), but can be confirmed for apply; once confirmed, they become the current run. When creating new [configuration versions](/terraform/cloud-docs/api-docs/configuration-versions) for saved plan runs, be sure to make them `provisional`. |

Reviewer: Small suggestion for active voice:
```suggestion
| `data.attributes.save-plan`                        | bool                 | false                                              | Specifies if this should be a saved plan run. Saved plan runs perform their plan and checks without becoming the workspace's current run (like a plan-only run), but will only become the workspace's current run if you confirm you want to apply it. When creating new [configuration versions](/terraform/cloud-docs/api-docs/configuration-versions) for saved plan runs, be sure to make them `provisional`. |
```
